### ISSUE TITLE
Prevent or warn when connecting incompatible features

### ISSUE THREAD

**Issue by boothym** [openstreetmap/iD#4173]
Recently saw a path connected to a power line and thought this might be something iD could prevent. Some features are not supposed to connect to each other, but end up connected likely due to snapping. Roads connecting to trees is another example.

It would be good if iD was able to prevent this, or at least warn users when saving.

**Comment by bhousel** [comment]
Yes, this is #2302, which I closed last year as being stale.  But I do think it's something we might be able to work into #4139, so let's leave it open as a maybe.

**Comment by boothym** [comment]
Saw this in someone's signature on the osm forum and reminded me of this issue 😝 
>There is only one place where you can connect a high voltage line to a river: on osm!

Of course it's difficult to know what the users are adding when they draw a line. Maybe in situations when a river or power line is being extended, it could only join onto something of its own "kind"? Just like if you disabled everything except the water features in the map features option when extending a river.

**Comment by andrewharvey** [comment]
anything with location=underground, or layer < 0 or tunnel=* should be prevented from snapping when drawing surface features eg. building=*

**Comment by matkoniecz** [comment]
I attempted to look into it. First step toward potential implementation is in https://github.com/openstreetmap/id-tagging-schema/pull/2454
